src/middlewared/middlewared/plugins/system.py

```diff
@@ -1014,6 +1014,39 @@ def run(self):
             )
 
 
+async def update_timeout_value(middleware, *args):
+    if not await middleware.call(
+        'tunable.query', [
+            ['var', '=', 'kern.init_shutdown_timeout'],
+            ['type', '=', 'SYSCTL'],
+            ['enabled', '=', True]
+        ]
+    ):
+        # Default 120 seconds is being added to scripts timeout to ensure other
+        # system related scripts can execute safely within the default timeout
+        timeout_value = 120 + sum(
+            list(
+                map(
+                    lambda i: i['timeout'],
+                    await middleware.call(
+                        'initshutdownscript.query', [
+                            ['enabled', '=', True],
+                            ['when', '=', 'SHUTDOWN']
+                        ]
+                    )
+                )
+            )
+        )
+
+        ret = await run(
```

Comment:
Why don't we use `sysctl` module here?

Reply:
I wasn't sure we could, i thought we used it only to retrieve sysctl values. I'll proceed with the module, thank you

Reply:
The reason I didnt note that is because this is a coroutine and sysctl is not aio friendly.
If we do that should probably use a thread since I am not sure if/how a sysctl set could block

Reply:
`await middleware.run_in_thread(lambda: setattr(sysctl.filter("...")[0], "value", ...))`